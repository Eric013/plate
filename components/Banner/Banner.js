export default () => (
  <div>
    <div className='banner'>
      <div className='header-main'>
        <h1>
          Task Management in the Most Simplistic Form, <br />
          Welcome to <i>Plate</i>!
        </h1>
      </div>
      <div className='header-secondary'>
        <h5>
          Create multiple plates and customize your very own task layout.<br /> Register now
          to begin managing your tasks!
        </h5>
      </div>
    </div>
    <style jsx>{`
      .banner {
        background-image: url('/static/plateGradient.png');
        height: 500px;
        padding-top: 80px;
      }
      .header-main h1 {
        text-align: center;
        color: white;
        font-size: 3.5em;
      }
      .header-secondary h5{
        text-align: center;
        color: white;
        font-style: italic;
        margin-top: 30px;
      }
      @media only screen and (max-width: 660px) {
        .banner {
          padding-top: 80px;
          padding-right: 40px;
          padding-left: 40px;
          height: 400px;
        }
         .header-main h1 {
           font-size: 1.5em;
         }
         .header-secondary h5 {
           font-size: 1.0em;
         }
      }
    `}</style>
  </div>
)
