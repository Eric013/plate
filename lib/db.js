import firebase from 'firebase'

// Initialize Firebase
const config = {
  apiKey: 'AIzaSyDL-jDWSC5lrRfRsngBu2gPTEmL2hO2SXE',
  authDomain: 'plate-a0b1b.firebaseapp.com',
  databaseURL: 'https://plate-a0b1b.firebaseio.com',
  storageBucket: 'plate-a0b1b.appspot.com',
  messagingSenderId: '1011831735527'
}

try {
  firebase.initializeApp(config)
} catch (err) {
  // we skip the 'already exists' message which is
  // not an actual error when we're hot-reloading
  if (!/already exists/.test(err.message)) {
    console.error('Firebase initialization error', err.stack)
  }
}

const db = firebase.database()

export default db
