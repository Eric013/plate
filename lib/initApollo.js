import ApolloClient, { createNetworkInterface } from 'apollo-client'

let apolloClient = null

// Cretae a new ApolloClient
function createClient (headers) {
  return new ApolloClient({
    ssrMode: !process.browser,
    dataIdFromObject: result => result.id || null,
    networkInterface: createNetworkInterface({
      uri: process.env.NODE_ENV === 'development' ? 'http://localhost:3000/graphql' : 'https://plate.now.sh/graphql',
      opts: {
        credentials: 'same-origin'
        // Pass headers here if your graphql server requires them
      }
    })
  })
}

// Initialize the new ApolloClient taking
// server vs client into account
export const initApollo = (headers) => {
  if (!process.browser) {
    return createClient(headers)
  }
  if (!apolloClient) {
    apolloClient = createClient(headers)
  }
  return apolloClient
}
