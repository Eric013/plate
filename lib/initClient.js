import { ApolloClient, createBatchingNetworkInterface } from 'react-apollo'

import Cookies from 'js-cookie'
import fetch from 'isomorphic-fetch'

let apolloClient = null

// Polyfill fetch() on the server (used by apollo-client)
if (!process.browser) {
  global.fetch = fetch
}

function _initClient(headers, initialState) {
  const networkInterface = createBatchingNetworkInterface({
    uri: process.env.NODE_ENV === 'development'
      ? 'http://localhost:3000/graphql'
      : 'https://plate.now.sh/graphql',
    batchInterval: 10,
    opts: {
      credentials: 'include'
    }
  })

  networkInterface.use([
    {
      applyBatchMiddleware(req, next) {
        /* eslint-disable no-param-reassign */
        if (!req.options.headers) {
          req.options.headers = {}
        }
        req.options.headers.authorization = Cookies.get('token') || null
        /* eslint-enable no-param-reassign */
        next()
      }
    }
  ])

  return new ApolloClient({
    initialState,
    networkInterface,
    queryDeduplication: true,
    ssrMode: !process.browser,
    dataIdFromObject: result => result.id || null,
    connectToDevTools: process.browser
  })
}

export const initClient = (headers, initialState = {}) => {
  if (!process.browser) {
    return _initClient(headers, initialState)
  }
  if (!apolloClient) {
    apolloClient = _initClient(headers, initialState)
  }
  return apolloClient
}
