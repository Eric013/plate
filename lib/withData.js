import rootReducer, { client } from '../redux/rootReducer'

import { ApolloProvider } from 'react-apollo'
import React from 'react'
import { initStore } from '../redux/createStore'

export default (Component) => (
  class extends React.Component {
    static getInitialProps ({ req }) {
      const isServer = !!req
      const store = initStore(rootReducer, {}, isServer)
      return { initialState: store.getState(), isServer }
    }

    constructor (props) {
      super(props)
      this.store = initStore(rootReducer, props.initialState, props.isServer)
    }

    render () {
      return (
        <ApolloProvider client={client} store={this.store}>
          <Component />
        </ApolloProvider>
      )
    }
  }
)
