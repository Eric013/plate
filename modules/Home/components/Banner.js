import BorderedButton from '../../../components/BorderedButton';
import Link from 'next/link';
import React from 'react';

const Banner = () => {
  return (
    <div>
      <div className="banner">
        <div className="header-main">
          <h1>
            Task Management in the Most Simplistic Form, <br />
            Welcome to <i>Plate</i>!
          </h1>
        </div>
        <div className="header-secondary">
          <h5>
            Create multiple plates and customize your very own task layout.
            <br />
            Register now
            to begin managing your tasks!
          </h5>
        </div>
        <div className="register-button">
          <Link prefetch href="/register">
            <a>
              <BorderedButton type="button" label="Register" />
            </a>
          </Link>
        </div>
      </div>
      <style jsx>
        {
          `
          .banner {
            background: linear-gradient(-90deg, rgb(141, 95, 115), #343f53);
            height: 500px;
            padding-top: 80px;
          }
          .header-main h1 {
            text-align: center;
            color: white;
            font-size: 3.5em;
          }
          .header-secondary h5{
            text-align: center;
            color: white;
            font-style: italic;
            margin-top: 30px;
          }
          .register-button {
            margin-top: 50px;
            display: flex;
            justify-content: center;
          }
          @media only screen and (max-width: 660px) {
            .banner {
              padding-top: 80px;
              padding-right: 40px;
              padding-left: 40px;
              height: 450px;
            }
            .header-main h1 {
              font-size: 1.5em;
            }
            .header-secondary h5 {
              font-size: 1.0em;
            }
          }
          `
        }
      </style>
    </div>
  );
};

export default Banner;
