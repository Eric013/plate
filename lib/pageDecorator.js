import './tap_events';
import 'isomorphic-fetch';

import { ApolloProvider, getDataFromTree } from 'react-apollo';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import React from 'react';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { initApollo } from './initApollo';
import { initStore } from '../store/initStore';

type Props = {
  headers?: Object,
  initialState?: Object,
  userAgent?: string,
};

export default Component => class extends React.Component {
  props: Props;

  static async getInitialProps(ctx) {
    const { req } = ctx;
    const headers = ctx.req ? ctx.req.headers : {};

    // Need the user agent for MuiThemeProvider when rendered on the
    // server vs client
    const userAgent = req ? req.headers['user-agent'] : navigator.userAgent;

    // Initialize apollo client
    const client = initApollo(headers);

    // Setup redux store
    const store = initStore(client, client.initialState);

    // Get inital props from the component passed in
    // if there are any
    let props = {};
    if (Component.getInitialProps) {
      props = await Component.getInitialProps(ctx);
    }

    // Were rendering on the server
    if (!process.browser) {
      const app = (
        <ApolloProvider client={client} store={store}>
          <MuiThemeProvider muiTheme={getMuiTheme({ userAgent: userAgent })}>
            <Component {...props} />
          </MuiThemeProvider>
        </ApolloProvider>
      );
      await getDataFromTree(app);
    }

    // Get the state from the store and return
    // out initial props
    const state = store.getState();
    return {
      initialState: {
        ...state,
      },
      headers,
      userAgent,
      ...props,
    };
  }

  constructor(props: Props) {
    super(props);
    this.client = initApollo(props.headers);
    this.store = initStore(this.client, props.initialState);
  }

  render() {
    return (
      <ApolloProvider client={this.client} store={this.store}>
        <MuiThemeProvider
          muiTheme={getMuiTheme({ userAgent: this.props.userAgent })}
        >
          <Component {...this.props} />
        </MuiThemeProvider>
      </ApolloProvider>
    );
  }
};
