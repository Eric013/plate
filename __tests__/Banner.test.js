/* global it, expect, describe */

import "../lib/tap_events";

import Banner from "../components/Banner/Banner";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import React from "react";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import { render } from "react-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  render(
    <MuiThemeProvider muiTheme={getMuiTheme()}>
      <Banner />
    </MuiThemeProvider>,
    div
  );
});

describe("Banner snapshot", () => {
  it('renders without crashing!"', () => {
    const component = renderer.create(
      <MuiThemeProvider muiTheme={getMuiTheme()}>
        <Banner />
      </MuiThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
