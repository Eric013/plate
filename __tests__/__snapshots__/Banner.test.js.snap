exports[`Banner snapshot renders without crashing!" 1`] = `
<div
  data-jsx={1531263508}>
  <div
    className="banner"
    data-jsx={1531263508}>
    <div
      className="header-main"
      data-jsx={1531263508}>
      <h1
        data-jsx={1531263508}>
        Task Management in the Most Simplistic Form, 
        <br
          data-jsx={1531263508} />
        Welcome to 
        <i
          data-jsx={1531263508}>
          Plate
        </i>
        !
      </h1>
    </div>
    <div
      className="header-secondary"
      data-jsx={1531263508}>
      <h5
        data-jsx={1531263508}>
        Create multiple plates and customize your very own task layout.
        <br
          data-jsx={1531263508} />
         Register now to begin managing your tasks!
      </h5>
    </div>
  </div>
</div>
`;
